packages/toolpad-app/src/components/CreateTreeItem.tsx

```diff
@@ -0,0 +1,151 @@
+import * as React from 'react';
+import { InputBase, alpha, Popover, Alert, useTheme, InputBaseProps } from '@mui/material';
+import { TreeItem, TreeItemProps, treeItemClasses } from '@mui/x-tree-view';
+
+interface CreateTreeItemProps extends Omit<TreeItemProps, 'nodeId' | 'label'> {
+  treeRef: React.RefObject<HTMLUListElement | null>;
+  open: boolean;
+  suggestedNewItemName?: string;
+  onCreate: (newItemName: string) => void | Promise<void>;
+  onClose: () => void | Promise<void>;
+  validateItemName?: (newItemName: string) => { isValid: boolean; errorMessage?: string };
+  inputProps?: Omit<
+    InputBaseProps,
+    'value' | 'onChange' | 'autoFocus' | 'onFocus' | 'onBlur' | 'onKeyDown' | 'fullWidth'
+  >;
+}
+
+export default function CreateTreeItem({
+  treeRef,
+  open,
+  suggestedNewItemName = '',
+  onCreate,
```

Comment:
Not sure how far to take this here. But this component is very focused on the "create" use case, while I'd expect the create logic to share 90% with the edit logic. Would an `EditableTreeItem` which is agnostic of whether it's in a create or edit scenario not make more sense?

Reply:
Should I try to make page names editable in the pages explorer maybe, to see how much of the logic I can share and how? And remove the ability to do it from the right-side panel? I guess that most of this logic might be sharable, but there is probably some more specific logic that I'd have to see how it fits.